### ISSUE TITLE
Aerospace units allowed a free turn every 8 hexes, regardless of velocity

### ISSUE THREAD

**Issue by TigerShark** [MegaMek/megamek#63]
(Version 0.41.16) In my recent test, aerospace units were allowed to turn safely after 8 hexes, regardless of their velocity. An Aero at velocity 3, for example, should only be able to safely turn at 16 hexes. They may turn at 8, but must use additional thrust to do so. Current build does not require them to expend additional thrust.

As a note, I had all of the StratOps rules for Aerospace movement/combat turned on during the first test. During the second, I unchecked all options to the default settings. The Aero behaved in this manner during both tests.

**Comment by TigerShark** [comment]
As a note, I had all of the StratOps rules for Aerospace movement/combat turned on during the first test. During the second, I unchecked all options to the default settings. The Aero behaved in this manner during both tests.

**Comment by pheonixstorm** [comment]
Happen to know the page reference? I have started reading up on aero but have not run across this yet. I have noticed as well about the turn radius.

**Comment by Dylan-M** [comment]
Curious, Aeros always behaved correctly in my games with them. At least for turns. Some of the maneuvers were wrong, but not the turns. We'll look into this.

**Comment by TigerShark** [comment]
Page 92, Total Warfare - "Straight Movement on Ground Maps Table"

Page 84 in the errata:
"On the ground map, each such change must be preceded by at least eight hexes of movement, 
which may be split across two turns"

**Comment by mkerensky** [comment]
I tested this a little more in 41.13, because I also thought it wasn't working right.  You do get into 'costly' turns if you increase the velocity up to 4-10 (not hard on the high altitude map, but a little ridiculous on most low altitude/ground maps)  Small Craft don't get the free turns nearly as easily, so I thought they were working fine. Dropships too.  After those tests, I thought ASF movement was working correctly as per an interpretation of the TW errata referenced above, but I liked it better before that was changed.

**Comment by arlith** [comment]
I need to spend more time reading the rules, but I think there's something with the tables between pg 84 and 92.

**Comment by HammerGS** [comment]
Just posting this here for easy of lookup.  Aero Movement received errata again in TW 4.0 errata

Link to errata:
https://dl.dropboxusercontent.com/u/357573/Total%20Warfare%20v4.2%202015-06-06.pdf

Below is the sections from errata with the errata'd text inserted:

Facing Changes
While operating in an atmosphere hex, fighters, aerodyne DropShips and aerodyne small craft need not spend Thrust Points to change facing. Instead, they use control surfaces built into
their wings to change facing and altitude. These surfaces allow the unit to make a number of free facing changes, depending on unit type and velocity. The faster a unit moves, the fewer free
facing changes it receives.

> “Before it can make a facing change on the low-altitude map, a unit must move in a straight line at least the number of hexes shown on the Straight Movement Table below. When moving on the ground map, multiply the minimum number of hexes required by eight. In both cases, this movement may be split across two turns.”

For example, an aerospace fighter traveling at Velocity 7 must move two hexes forward before making a facing change. A conventional fighter at the same effective velocity would need
to move only one hex between facing changes. An aerodyne DropShip would need to move three hexes. Regardless of velocity, a unit can only make one facing change one side in each hex.

Additional Facing Changes: Aerospace fighters and aerodyne DropShips and small craft can also spend Thrust Points to gain additional facing changes. At any point in their movement, such
units may spend Thrust Points equal to half their current velocity (round up) to change facing one hexside. 

> “However, they may not do so if they have already changed facing in that hex or if this is the first hex of their movement on the low-altitude map. On the ground map, each such change must be preceded by at least eight hexes of movement, which may be split across two turns.”

For example, the pilot of an aerospace fighter with a Velocity of 7 wants to make an additional facing change. This will cost 4 MP (half the fighter’s current velocity, rounded up).

Fixed-Wing Support Vehicles: These units must move double the minimum straight movement requirement of a conventional fighter.

Rolling Maneuvers: Units in atmosphere cannot roll simply by spending Thrust Points. They must instead perform the maneuver as a half-roll (see the Special Maneuvers Table, p. 85).

**Comment by TigerShark** [comment]
This was working during earlier versions of MM. I'm trying to track down which commit 'broke' the Aero movement.

**Comment by arlith** [comment]
It's not "broken", but more everything is following the table on pg 84.  I the commit is a6b926f50, which was added as a fix to a bug report about the latest errata.  I still haven't had time to read the rules, but my impression is that we're using the same code to implement two tables which do slightly different things.  My impression was that the table on pg 84 is for "free facing changes", and the table on pg 92 is "minimum movement  before making a turn."  I think that MM is properly applying the table on pg 84, but it's also using that same table to determine minimum moves before a facing chance.

**Comment by TigerShark** [comment]
It appears the turns are functioning properly once that commit is rolled back.

**Comment by arlith** [comment]
Yes, but then you aren't getting free turns soon enough, according to the table on pg 84, which is what that commit fixes.

**Comment by TigerShark** [comment]
Page 84, "While operating in a atmosphere hex, fi ghters, aerodyne DropShips and aerodyne small craft need not spend Thrust Points to change facing." 

This section's table pertains to Atmosphere maps, not Ground-level maps. High-Altitude and Low-Altitude maps are different than Ground maps.

**Comment by arlith** [comment]
The paragraph just after that, on TO pg 84 says:

> Before it can make a facing change on the low-altitude map,
> a unit must move in a straight line at least the number of hexes
> shown on the Straight Movement Table below. When moving on
> the ground map, multiply the minimum number of hexes required
> by eight.

Then, in the paragraph after that, in the subsection **Additional Facing Changes**, it says:
>Aerospace fighters and aerodyne DropShips and small craft can also spend Thrust Points to gain additional facing changes.
...
>On the ground map, each [facing] change must be preceded by at least eight
hexes of movement.

There are three types of non-space map for Aeros: the High altitude map, the low altitude map, and the ground map.  I think when TW says "operating in an atmosphere hex" it's referring to any hex base the space-atmosphere interface in the high altitude map, so this would include all hexes in the low altitude and ground maps.

Basically, the text/table on page 84 is in direct contradiction to what is in the text/table on page 92.  When I "fixed" the issue previously, someone had been pointing to the text/table on pg 84 and I had forgotten about the table on pg 92 (why would I recall a second table depicting the same thing?).  Currently, MM is implemented exactly according to what is stated on pg 84.  I guess I'll have to get a rules clarification or something...

**Comment by arlith** [comment]
Looks like this issue has been taken care of in errata that hasn't made it into official published errata yet: http://bg.battletech.com/forums/index.php?topic=50094.0

The line that I quoted above that makes it look like the table on pg 84 should apply to aeros on the ground map, has been removed.  So now it's clear that the table on pg 92 should apply.

**Comment by arlith** [comment]
Fixed in c45f9f1.
